
Fix build with kernel 3.12

Signed-off-by: Thomas Backlund <tmb@mageia.org>

--- 13.11-beta1/common/lib/modules/fglrx/build_mod/firegl_public.h.orig	2013-10-04 18:15:41.000000000 +0300
+++ 13.11-beta1/common/lib/modules/fglrx/build_mod/firegl_public.h	2013-10-13 19:29:58.828619702 +0300
@@ -265,7 +265,11 @@ typedef struct {
  */
 typedef int             KCL_TYPE_Pid;       /** Process identifier */
 typedef int             KCL_TYPE_Tgid;      /** Thread Group identifier */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 12, 0)
 typedef int             KCL_TYPE_Uid;       /** User identifier */
+#else
+typedef kuid_t          KCL_TYPE_Uid;       /** User identifier */
+#endif
 
 /** Atomic variables
  * This type is defined using struct to make sure compiled code will
