--- beta11/common/lib/modules/fglrx/build_mod/firegl_public.c.orig	2012-12-03 04:03:45.000000000 +0200
+++ beta11/common/lib/modules/fglrx/build_mod/firegl_public.c	2012-12-07 03:11:08.000000000 +0200
@@ -3892,7 +3892,11 @@ int ATI_API_CALL KCL_MEM_VM_MapRegion(KC
                 KCL_DEBUG_ERROR(REMAP_PAGE_RANGE_STR " failed\n");
                 return -EAGAIN;
             }
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
             vma->vm_flags |= VM_SHM | VM_RESERVED; /* Don't swap */
+#else
+            vma->vm_flags |= VM_SHM | VM_DONTEXPAND | VM_DONTDUMP; /* Don't swap */
+#endif
             vma->vm_ops = &vm_ops;
 			break;
 
@@ -3922,14 +3926,22 @@ int ATI_API_CALL KCL_MEM_VM_MapRegion(KC
                 KCL_DEBUG_ERROR(REMAP_PAGE_RANGE_STR " failed\n");
                 return -EAGAIN;
             }
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
             vma->vm_flags |= VM_SHM | VM_RESERVED; /* Don't swap */
+#else
+            vma->vm_flags |= VM_SHM | VM_DONTEXPAND | VM_DONTDUMP; /* Don't swap */
+#endif
             vma->vm_ops = &vm_ops;
             }
 			break;
 #endif                    
 
         case __KE_SHM:
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
             vma->vm_flags |= VM_SHM | VM_RESERVED; /* Don't swap */
+#else
+            vma->vm_flags |= VM_SHM | VM_DONTEXPAND | VM_DONTDUMP; /* Don't swap */
+#endif
             vma->vm_ops = &vm_shm_ops;
             break;
 
@@ -3937,7 +3949,11 @@ int ATI_API_CALL KCL_MEM_VM_MapRegion(KC
 
             pages = (vma->vm_end - vma->vm_start) >> PAGE_SHIFT;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
             vma->vm_flags |= VM_RESERVED;
+#else
+            vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;
+#endif
 
             //vma->vm_flags |=  VM_SHM | VM_LOCKED; /* DDDDDDDDDDon't swap */
             //vma->vm_mm->locked_vm += pages; /* Kernel tracks aqmount of locked pages */
@@ -3946,14 +3962,22 @@ int ATI_API_CALL KCL_MEM_VM_MapRegion(KC
 
         case __KE_CTX:
             pages = (vma->vm_end - vma->vm_start) >> PAGE_SHIFT;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
             vma->vm_flags |= VM_LOCKED | VM_SHM | VM_RESERVED; /* Don't swap */
+#else
+            vma->vm_flags |= VM_LOCKED | VM_SHM | VM_DONTEXPAND | VM_DONTDUMP; /* Don't swap */
+#endif
             vma->vm_mm->locked_vm += pages; /* Kernel tracks aqmount of locked pages */
             vma->vm_ops = &vm_ctx_ops;
             break;
 
         case __KE_PCI_BQS:
             pages = (vma->vm_end - vma->vm_start) >> PAGE_SHIFT;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
             vma->vm_flags |= VM_LOCKED | VM_SHM | VM_RESERVED; /* Don't swap */
+#else
+            vma->vm_flags |= VM_LOCKED | VM_SHM | VM_DONTEXPAND | VM_DONTDUMP; /* Don't swap */
+#endif
             vma->vm_mm->locked_vm += pages; /* Kernel tracks aqmount of locked pages */
             vma->vm_ops = &vm_pci_bq_ops;
             break;
@@ -3984,9 +4008,17 @@ int ATI_API_CALL KCL_MEM_VM_MapRegion(KC
                     return -EAGAIN;
                 }
 #ifdef __x86_64__
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
                 vma->vm_flags |= VM_RESERVED;
 #else
+                vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;
+#endif
+#else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
                 vma->vm_flags |= VM_SHM | VM_RESERVED; /* Don't swap */
+#else
+                vma->vm_flags |= VM_SHM | VM_DONTEXPAND | VM_DONTDUMP; /* Don't swap */
+#endif
 #endif
                 vma->vm_ops = &vm_ops;
             }
@@ -4015,9 +4047,17 @@ int ATI_API_CALL KCL_MEM_VM_MapRegion(KC
                     return -EAGAIN;
                 }
 #ifdef __x86_64__
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
                 vma->vm_flags |= VM_RESERVED;
 #else
+                vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;
+#endif
+#else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
                 vma->vm_flags |= VM_SHM | VM_RESERVED; /* Don't swap */
+#else
+                vma->vm_flags |= VM_SHM | VM_DONTEXPAND | VM_DONTDUMP; /* Don't swap */
+#endif
 #endif
                 vma->vm_ops = &vm_agp_bq_ops;
             }
@@ -4025,7 +4065,11 @@ int ATI_API_CALL KCL_MEM_VM_MapRegion(KC
 #endif /* __AGP__BUILTIN__ */
 
         case __KE_KMAP:
-		    vma->vm_flags |= VM_SHM | VM_RESERVED;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
+	    vma->vm_flags |= VM_SHM | VM_RESERVED;
+#else
+	    vma->vm_flags |= VM_SHM | VM_DONTEXPAND | VM_DONTDUMP;
+#endif
             vma->vm_ops = &vm_kmap_ops;
             if (readonly && (vma->vm_flags & VM_WRITE))
             {
@@ -4046,7 +4090,11 @@ int ATI_API_CALL KCL_MEM_VM_MapRegion(KC
 #endif            
             // fall through
          case __KE_GART_CACHEABLE:
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,7,0)
              vma->vm_flags |= VM_RESERVED;
+#else
+             vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;
+#endif
              vma->vm_ops = &vm_gart_ops;
              break;
         default:
