diff -Nurp fglrx-9.010.8/common/lib/modules/fglrx/build_mod/kcl_acpi.c fglrx-9.010.8-fix-3.8/common/lib/modules/fglrx/build_mod/kcl_acpi.c
--- fglrx-9.010.8/common/lib/modules/fglrx/build_mod/kcl_acpi.c	2012-12-03 04:03:45.000000000 +0200
+++ fglrx-9.010.8-fix-3.8/common/lib/modules/fglrx/build_mod/kcl_acpi.c	2013-01-10 21:16:53.316407229 +0200
@@ -775,7 +775,9 @@ static unsigned int KCL_ACPI_SearchHandl
 unsigned int ATI_API_CALL KCL_ACPI_GetHandles(kcl_match_info_t *pInfo)
 {
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,12)
-    #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,19)
+    #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0)
+        pInfo->video_handle = pInfo->pcidev->dev.acpi_node.handle;
+    #elif LINUX_VERSION_CODE > KERNEL_VERSION(2,6,19)
         pInfo->video_handle = pInfo->pcidev->dev.archdata.acpi_handle;
     #else 
         pInfo->video_handle = pInfo->pcidev->dev.firmware_data;
